kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/organizationalunit/contributors/tab/ContributorsListItemPresenter.java

```diff
@@ -171,7 +171,7 @@ private Promise<Boolean> isValid(final Contributor contributor,
                 return promises.resolve(false);
             }
 
-            final boolean validUsername = !contributorsListService.requireValidUsername() || parentPresenter.getValidUsernames().contains(contributor.getUsername());
+            final boolean validUsername = !contributorsListService.requireValidUsername() && parentPresenter.getValidUsernames().contains(contributor.getUsername());
```

Comment:
```suggestion
            final boolean validUsername = !contributorsListService.requireValidUsername() || parentPresenter.getValidUsernames().contains(contributor.getUsername());
```

This has to be a `||` because the `requireValidUsername` method informs if the new username has to be validated againt a list of names (project contributors have to be space contributors). Which means that if the service does not require that the username belongs to a list of valid names, it is valid. And if the service require that the username belongs to a list of valid names, then it checks the second condition.

Reply:
Hi @paulovmr , While adding the contributors to the space, I can see a it shows the list of All Users. Shouldn't this be validated against all users of RHPAM?

Reply:
I still don't see a problem with the current code on main for this condition. Unless there is a bug which I don't know of related to this line, I don't think it is wrong as per my first comment, and is not on the scope of this JIRA. This condition is what guarantees that a project contributor is a space contributor.

Reply:
Hi @paulovmr , Sorry, I misunderstood the Contributor scope. I thought that space users should be validated against all the Users. I'm reverting these changes and retaining the SafeHTML-based fix. Also, as @domhanak raised concerns over the security screen XSS vulnerability, I am raising a new PR for that as well.
^ @Rikkola

Reply:
I think this SafeHTML-based fix should be applied to the username the moment it is saved instead of when it is displayed, as there might be other places that also display it.

Reply:
With https://github.com/kiegroup/appformer/pull/1379 The username with XSS content can not be saved, so I think it is being shown only on the Contributors tab. I we apply it at the time username is saved, then it will be saved as encoded string replacing `<, >, &, ' and "`.

Reply:
In this case could you add the same validation in the appformer PR to the [isValid](https://github.com/kiegroup/kie-wb-common/blob/main/kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/organizationalunit/contributors/tab/ContributorsListItemPresenter.java#L164) as a new validation?

Reply:
Done 👍  I have updated the changes.